compiler-k2/build.gradle.kts

```diff
@@ -23,13 +23,18 @@ dependencies {
   api(project(":compiler-k2-api"))
   api(libs.kotlin.compiler)
 
+  implementation(project(":compiler-utils"))
+  implementation(libs.dagger2)
```

Comment:
If we just use `ClassId.joinSimpleNames(...)` and define the FqNames using strings, we can drop both these dependencies.